Sources/MagicSDK/Core/Relayer/WebViewController.swift

```diff
@@ -143,6 +151,23 @@ class WebViewController: UIViewController, WKUIDelegate, WKScriptMessageHandler,
             throw RpcProvider.ProviderError.invalidJsonResponse(json: payloadStr)
         }
     }
+    
+    private func makeProductAnnouncement(payloadStr: String) throws -> Void {
+        // Decoding the JSON string into the Payload struct
+        guard let data = payloadStr.data(using: .utf8) else {
+            throw NSError(domain: "Invalid payload string", code: 1001, userInfo: nil)
```

Comment:
I think we may silently catch the error, as the developer doesn't need to handle this